src/main/java/com/commercetools/sync/commons/helpers/BaseReferenceResolver.java

```diff
@@ -48,69 +42,23 @@ protected BaseReferenceResolver(@Nonnull final S options) {
     /**
      * This method fetches the id value on the passed {@link ResourceIdentifier}, if valid. If it is not valid, a
      * {@link ReferenceResolutionException} will be thrown. The validity checks are:
-     * <ol>
-     * <li>Checks if the id value has a UUID format and the {@code allowUuidKeys} flag set to true, or the id value
-     * doesn't have a UUID format.</li>
+     * <ul>
      * <li>Checks if the id value is not null or not empty.</li>
-     * </ol>
-     * If the above checks pass, the id value is returned. Otherwise a {@link ReferenceResolutionException} is thrown.
-     *
-     * <p>Note: Currently the key is expected to be set on the id field of the {@link ResourceIdentifier}. This is in
-     * order to make key setting consistent on both {@link io.sphere.sdk.models.Reference} and
-     * {@link ResourceIdentifier}. However, this will be changed when GITHUB ISSUE#138 is resolved.
+     * </ul>
+     * If the above checks pass, the key value is returned. Otherwise a {@link ReferenceResolutionException} is thrown.
      *
-     * @param resourceIdentifier the reference from which the id value is validated and returned.
-     * @param allowUuidKeys flag that signals whether the key could be UUID format or not.
-     * @return the id value on the {@link ResourceIdentifier}
+     * @param resourceIdentifier the reference from which the key value is validated and returned.
+     * @return the key value on the {@link ResourceIdentifier}
      * @throws ReferenceResolutionException if any of the validation checks fail.
      */
     @Nonnull
-    @SuppressWarnings("ConstantConditions") //To remove the warning on the key is null, because it can't be null.
-    protected static String getKeyFromResourceIdentifier(@Nonnull final ResourceIdentifier resourceIdentifier,
-                                                         final boolean allowUuidKeys)
+    protected static String getKeyFromResourceIdentifier(@Nonnull final ResourceIdentifier resourceIdentifier)
         throws ReferenceResolutionException {
```

Comment:
is this method is needed at all now?

Reply:
yes, it is still needed. Because the way the key is fetched from a resource identifier is still not normal - we presume that the key value will be placed in the id field. This will be changed in the next PR addressing: #138 

Not only that, but the method also validates such key. Right now it's only that it is not blank. But this validation was more than that before and could be changed also in the future. So I would rather leave it in the method.

Reply:
yes, it is still needed. Because the way the key is fetched from a resource identifier is still not normal - we presume that the key value will be placed in the id field. This will be changed in the next PR addressing: #138 

Not only that, but the method also validates such key. Right now it's only that it is not blank. But this validation was more than that before and could be changed also in the future. So I would rather leave it in the method.

Reply:
ok